src/engine/ParseLineCommand.java

```diff
@@ -0,0 +1,60 @@
+package engine;
+
+import data.LineData;
+import item.ILine;
+import utils.SortType;
+import utils.Utils;
+
+import java.util.*;
+
+
+public class ParseLineCommand extends Command<LineData> {
+    public ParseLineCommand(LineData input, SortType sortType) {
+        super(input, sortType);
+    }
+
+    @Override
+    public String execute() {
+        List<ILine> data = getInput().getData();
+        List<String> result;
+        if (SortType.BY_COUNT.equals(getSortType())) {
+            result = sortByCount(data);
+        } else {
+            result = sortByNatural(data);
+        }
+        String total = String.format("Total lines: %d.", data.size());
+        String sort = String.join("\n", result);
+        return String.format("%s\n%s", total, sort);
+    }
+
+    private List<String> sortByNatural(List<ILine> data) {
```

Comment:
It is a good example when to use comparable
data.sort(Comparator.naturalOrder()) will work for all types